src/components/SplashScreen.tsx

```diff
@@ -1,51 +1,72 @@
 import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
 import styles from "./SplashScreen.module.css";
 import Circle from "./splashScreen/Circle";
 import TitleBadge from "./splashScreen/TitleBadge";
 
 type SplashScreenProps = {
   shouldTransition: boolean;
-};
-
-const splashScreenDisplayOffStyle = {
-  display: "none",
+  defaultPos: boolean;
+  title?: string;
+  actionDesc?: string;
 };
 
 export default function SplashScreen(props: SplashScreenProps) {
-  const [shouldDisplay, setShouldDisplay] = useState(true);
   // delay and this state are neccessary for transition applied on route transition, so the animation still happens.
+  const [isDefaultPositionOn, setIsDefaultPositionOn] = useState(props.defaultPos);
   const [isTransitionReady, setIsTransitionReady] = useState(false);
+  const navigate = useNavigate();
+  const handleButtonAction = () => {
+    setIsTransitionReady(true);
+  };
+
+  // transition control useEffect.
   useEffect(() => {
     const handleTransition = (transition: boolean) => {
       transition && isTransitionReady !== transition && setIsTransitionReady(transition);
     };
-    const toggleDisplay = () => {
-      setShouldDisplay((prevVal) => !prevVal);
-    };
     const transitionTimeoutId = setTimeout(() => handleTransition(props.shouldTransition), 1);
-    const displayTimeoutId = setTimeout(() => toggleDisplay(), 4000);
     return () => {
       clearTimeout(transitionTimeoutId);
-      clearTimeout(displayTimeoutId);
     };
   }, [props.shouldTransition]);
+
+  // navigation and default position control useEffect.
+  useEffect(() => {
```

Comment:
**nitpick:** I would move this logic up to `App.tsx`, and not make it with `useEffect` but with simple `handleClick` where you set `isAnimation=true` . Currently your solution hardcodes navigation to the specific route, and you are able to parametrize it to make it more flexible.